tsdb/exemplar.go

```diff
@@ -169,6 +175,18 @@ func (ce *CircularExemplarStorage) AddExemplar(l labels.Labels, e exemplar.Exemp
 	ce.lock.Lock()
 	defer ce.lock.Unlock()
 
+	// Exemplar label length does not include chars involved in text rendering such as quotes
+	// equals sign, or commas. See definiton of const ExemplarMaxLabelLength.
+	labelSetLen := 0
+	for _, l := range e.Labels {
+		labelSetLen += len(l.Name)
+		labelSetLen += len(l.Value)
```

Comment:
We should perf test that.

I would be fine to either take the approach here or to multiply the value by 4. Just like we do when we ingest openmetrics, we do not aim at digging into every corner case. Performance must also be taken into consideration.

Reply:
We can be permissive here and just allow anything up to 512 bytes, but the problem is that we would then potentially accept non-compliant exemplars. A remote-write receiver that is less permissive could reject those.

Perhaps it's worth raising an issue with OpenMetrics to reconsider.

Reply:
https://github.com/OpenObservability/OpenMetrics/issues/198
Although that might just add another one to the pile of rejected issues of mine…

Reply:
I would note that currently we allow any size. Even 512 bytes would be an improvement.

Reply:
But, as fittingly commented on my issue, it would still render Prometheus as not compliant with OpenMetrics.

At least now I cannot claim anymore that OpenMetrics puts the burden on the side of the producer (which I find a fundamental flaw of the design). At least it makes it hard for the consumer, too.